### ISSUE TITLE
`semicolon-inside-block` does not like `try_blocks`

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#16696]
Using the following flags

`--force-warn clippy::semicolon-inside-block`

this code:

```rust
//@ pp-exact
//@ edition: 2024

#![feature(try_blocks, try_blocks_heterogeneous)]

fn main() { try { Some(1)? }; try bikeshed Result<u32, ()> { 3 }; }

```

caused the following diagnostics:

```
    Checking _1769958047167023841 v0.1.0 (/tmp/icemaker_global_tempdir.gPgWiqot4T9Z/icemaker_clippyfix_tempdir.1EYX4e07DjKg/_1769958047167023841)
warning: consider moving the `;` inside the block for consistent formatting
 --> src/main.rs:6:13
  |
6 | fn main() { try { Some(1)? }; try bikeshed Result<u32, ()> { 3 }; }
  |             ^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#semicolon_inside_block
  = note: requested on the command line with `--force-warn clippy::semicolon-inside-block`
help: put the `;` here
  |
6 - fn main() { try { Some(1)? }; try bikeshed Result<u32, ()> { 3 }; }
6 + fn main() { try { Some(1)?; } try bikeshed Result<u32, ()> { 3 }; }
  |

warning: `_1769958047167023841` (bin "_1769958047167023841") generated 1 warning (run `cargo clippy --fix --bin "_1769958047167023841" -p _1769958047167023841 -- -Aclippy::all -Awarnings --force-warn clippy::semicolon-inside-block --cap-lints warn` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.15s

```

However after applying these diagnostics, the resulting code:

```rust
//@ pp-exact
//@ edition: 2024

#![feature(try_blocks, try_blocks_heterogeneous)]

fn main() { try { Some(1)?.expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }

```
no longer compiled:

```
    Checking _1769958047167023841 v0.1.0 (/tmp/icemaker_global_tempdir.gPgWiqot4T9Z/icemaker_clippyfix_tempdir.1EYX4e07DjKg/_1769958047167023841)
warning: failed to automatically apply fixes suggested by rustc to crate `_1769958047167023841`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0308]: mismatched types
 --> src/main.rs:6:19
  |
6 | fn main() { try { Some(1)?.expect("REASON").expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }
  |                   ^^^^^^^^ expected `()`, found `Option<_>`
  |
  = note: expected unit type `()`
                  found enum `std::option::Option<_>`
help: consider using `Option::expect` to unwrap the `std::option::Option<_>` value, panicking if the value is an `Option::None`
  |
6 | fn main() { try { Some(1)?.expect("REASON").expect("REASON").expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }
  |                           +++++++++++++++++

error[E0599]: no method named `expect` found for type `{integer}` in the current scope
 --> src/main.rs:6:28
  |
6 | fn main() { try { Some(1)?.expect("REASON").expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }
  |                            ^^^^^^ method not found in `{integer}`

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0308, E0599.
For more information about an error, try `rustc --explain E0308`.
Original diagnostics will follow.

error[E0308]: mismatched types
 --> src/main.rs:6:19
  |
6 | fn main() { try { Some(1)?.expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }
  |                   ^^^^^^^^ expected `()`, found `Option<_>`
  |
  = note: expected unit type `()`
                  found enum `std::option::Option<_>`
help: consider using `Option::expect` to unwrap the `std::option::Option<_>` value, panicking if the value is an `Option::None`
  |
6 | fn main() { try { Some(1)?.expect("REASON").expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }
  |                           +++++++++++++++++

error[E0599]: no method named `expect` found for type `{integer}` in the current scope
 --> src/main.rs:6:28
  |
6 | fn main() { try { Some(1)?.expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }
  |                            ^^^^^^ method not found in `{integer}`

Some errors have detailed explanations: E0308, E0599.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `_1769958047167023841` (bin "_1769958047167023841") due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
warning: failed to automatically apply fixes suggested by rustc to crate `_1769958047167023841`

after fixes were automatically applied the compiler reported errors within these files:

  * src/main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust-clippy/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
error[E0308]: mismatched types
 --> src/main.rs:6:19
  |
6 | fn main() { try { Some(1)?.expect("REASON").expect("REASON").expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }
  |                   ^^^^^^^^ expected `()`, found `Option<_>`
  |
  = note: expected unit type `()`
                  found enum `std::option::Option<_>`
help: consider using `Option::expect` to unwrap the `std::option::Option<_>` value, panicking if the value is an `Option::None`
  |
6 | fn main() { try { Some(1)?.expect("REASON").expect("REASON").expect("REASON").expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }
  |                           +++++++++++++++++

error[E0599]: no method named `expect` found for type `{integer}` in the current scope
 --> src/main.rs:6:28
  |
6 | fn main() { try { Some(1)?.expect("REASON").expect("REASON").expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }
  |                            ^^^^^^ method not found in `{integer}`

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0308, E0599.
For more information about an error, try `rustc --explain E0308`.
Original diagnostics will follow.

error[E0308]: mismatched types
 --> src/main.rs:6:19
  |
6 | fn main() { try { Some(1)?.expect("REASON").expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }
  |                   ^^^^^^^^ expected `()`, found `Option<_>`
  |
  = note: expected unit type `()`
                  found enum `std::option::Option<_>`
help: consider using `Option::expect` to unwrap the `std::option::Option<_>` value, panicking if the value is an `Option::None`
  |
6 | fn main() { try { Some(1)?.expect("REASON").expect("REASON").expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }
  |                           +++++++++++++++++

error[E0599]: no method named `expect` found for type `{integer}` in the current scope
 --> src/main.rs:6:28
  |
6 | fn main() { try { Some(1)?.expect("REASON").expect("REASON"); } try bikeshed Result<u32, ()> { 3 }; }
  |                            ^^^^^^ method not found in `{integer}`

error: could not compile `_1769958047167023841` (bin "_1769958047167023841" test) due to 2 previous errors

```

Version:
```
rustc 1.96.0-nightly (e3d66fe39 2026-03-07)
binary: rustc
commit-hash: e3d66fe39ae70380fa2365c008e2927479114844
commit-date: 2026-03-07
host: x86_64-unknown-linux-gnu
release: 1.96.0-nightly
LLVM version: 22.1.0

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->
